src/ast/semantic_analyser.cpp

```diff
@@ -800,6 +800,9 @@ void SemanticAnalyser::visit(Binop &binop)
   bool lsign = binop.left->type.is_signed;
   bool rsign = binop.right->type.is_signed;
 
+  if (is_context(binop.left) || is_context(binop.right))
+    bpftrace_.error(err_, binop.left->loc + binop.right->loc, "Calculation to context is not supported");
```

Comment:
This message is a bit vague, I do not have a good suggestion at the moment though. Maybe `pointer arithmetic on context is not allowed`.